File: packages/android/src/main/java/com/kozmos/components/Grid/Grid.figma.kt
package com.kozmos.components.grid

import androidx.compose.foundation.lazy.grid.items
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.unit.dp
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=359-1574")
class KozmosGridConnect {
    @FigmaProperty(FigmaType.Enum, "Columns")
    val columns: Int = Figma.mapping(
        "1" to 1,
        "2" to 2,
        "3" to 3,
        "4" to 4
    )

    @FigmaProperty(FigmaType.Enum, "Gap")
    val gap: Int = Figma.mapping(
        "2" to 8,
        "4" to 16,
        "6" to 24
    )

    @Composable
    fun ComponentExample() {
        KozmosGrid(
            cols = columns,
            spacing = gap.dp
        ) {
            items(listOf("One", "Two", "Three", "Four")) { label ->
                Text(label)
            }
        }
    }
}

File: packages/ios/Sources/Components/Grid/Grid.figma.swift
import SwiftUI
import Figma

struct KozmosGridConnect: FigmaConnect {
    let component = Grid<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=359-1574"

    @FigmaEnum(
        "Columns",
        mapping: [
            "1": 1,
            "2": 2,
            "3": 3,
            "4": 4
        ]
    )
    var columns: Int = 2

    @FigmaEnum(
        "Gap",
        mapping: [
            "2": CGFloat(8),
            "4": CGFloat(16),
            "6": CGFloat(24)
        ]
    )
    var spacing: CGFloat = 16

    var body: some View {
        Grid(cols: self.columns, spacing: self.spacing) {
            AnyView(Color.clear.frame(height: 48))
            AnyView(Color.clear.frame(height: 48))
            AnyView(Color.clear.frame(height: 48))
            AnyView(Color.clear.frame(height: 48))
        }
    }
}
